src/web/app/components/question-types/question-statistics/question-statistics-calculation/msq-question-statistics-calculation.ts

```diff
@@ -91,19 +91,29 @@ export class MsqQuestionStatisticsCalculation
     const perRecipientResponse: Record<string, Record<string, number>> = {};
     const recipientToTeam: Record<string, string> = {};
     const recipientEmails: Record<string, string> = {};
+    const recipientNames: Record<string, string> = {};
     for (const response of this.responses) {
-      perRecipientResponse[response.recipient] = perRecipientResponse[response.recipient] || {};
-      recipientEmails[response.recipient] = recipientEmails[response.recipient] || response.recipientEmail || '';
+      if (!response.recipientEmail) {
+        continue;
+      }
+      perRecipientResponse[response.recipientEmail] = perRecipientResponse[response.recipientEmail] || {};
+      const responseEmail = response.recipientEmail;
```

Comment:
Lets move this to the start of the for loop since it is used multiple times before this line as well